ultravox/data/datasets.py

```diff
@@ -1060,43 +1068,73 @@ def create_dataset(name: str, args: VoiceDatasetArgs) -> data.IterableDataset:
 
 
 class InterleaveDataset(data.IterableDataset):
-    """Interleaves multiple IterableDataset objects."""
+    """Interleaves multiple IterableDataset objects based on normalized weights."""
 
     def __init__(
-        self, datasets: Sequence[data.IterableDataset], repeat: bool = False
+        self,
+        datasets: Sequence[data.IterableDataset],
+        stop_strategy: str = "last_exhausted",
+        seed: Optional[int] = 42,
+        static: bool = False,
```

Comment:
I don't think we need this for anything other than unit tests, right? Accordingly I think we should just use a fixed seed in the test.

Reply:
I am not sure about this. Static removes random sampling and uses a static order, which could also be helpful for debugging, or for something we don't want any randomness, e.g., validation and evaluation.

Reply:
Understood, but in that case I tend to think it should be a separate class, rather than having a single class with two distinct code paths.